lib/pliny/librato/metrics/backend.rb

```diff
@@ -54,21 +53,13 @@ def start_timer
           end
         end
 
-        def start_counter
-          @counter = Thread.new do
-            loop do
-              msg = metrics_queue.pop
-              msg == POISON_PILL ? break : enqueue_librato(msg)
-            end
-          end
-        end
-
-        def enqueue_librato(msg)
-          sync { librato_queue.add(msg) }
-        end
-
         def flush_librato
-          sync { librato_queue.submit }
+          sync do
+            counter_cache.flush_to(librato_queue)
+            librato_queue.merge!(aggregator)
```

Comment:
It looks like `merge!` also submits. I think this should be `librato_queue.merge!(aggregator) unless aggregator.empty?`